be/src/simd/selector.h

```diff
@@ -216,6 +219,157 @@ inline void avx2_select_if_common_implement(uint8_t*& selector, T*& dst, const T
+                }
+
+                uint16x8_t vec_b;
+                if constexpr (!right_const) {
+                    vec_b = vld1q_u16(reinterpret_cast<const uint16_t*>(b) + i * 8);
+                } else {
+                    vec_b = vdupq_n_u16(*reinterpret_cast<const uint16_t*>(b));
+                }
+
+                // Convert first 8 uint8 masks to uint16 masks using lookup table, effectively duplicating each uint8
+                uint8x16_t index = {0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7};
+                uint8x16_t mask = vqtbl1q_u8(loaded_mask, index);
+
+                // Select result based on mask
+                uint16x8_t result = vbslq_u16(vreinterpretq_u16_u8(mask), vec_a, vec_b);
+
+                vst1q_u16(reinterpret_cast<uint16_t*>(dst) + i * 8, result);
+                loaded_mask = vextq_u8(loaded_mask, loaded_mask, 8);
+            }
+        } else if constexpr (data_size == 4) { // int32/float
+            // Process 4 groups, each handling 4 int32
+            for (int i = 0; i < 4; i++) {
+                uint32x4_t vec_a;
+                if constexpr (!left_const) {
+                    vec_a = vld1q_u32(reinterpret_cast<const uint32_t*>(a) + i * 4);
+                } else {
+                    vec_a = vdupq_n_u32(*reinterpret_cast<const uint32_t*>(a));
+                }
+
+                uint32x4_t vec_b;
+                if constexpr (!right_const) {
+                    vec_b = vld1q_u32(reinterpret_cast<const uint32_t*>(b) + i * 4);
+                } else {
+                    vec_b = vdupq_n_u32(*reinterpret_cast<const uint32_t*>(b));
+                }
+
+                uint8x16_t index = {0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3};
+                uint8x16_t mask = vqtbl1q_u8(loaded_mask, index);
+
+                uint32x4_t result = vbslq_u32(vreinterpretq_u32_u8(mask), vec_a, vec_b);
+
+                vst1q_u32(reinterpret_cast<uint32_t*>(dst) + i * 4, result);
+                loaded_mask = vextq_u8(loaded_mask, loaded_mask, 4);
+            }
+        } else if constexpr (data_size == 8) { // int64/double
+            // Process 8 groups, each handling 2 int64
+            for (int i = 0; i < 8; i++) {
+                uint64x2_t vec_a;
+                if constexpr (!left_const) {
+                    vec_a = vld1q_u64(reinterpret_cast<const uint64_t*>(a) + i * 2);
+                } else {
+                    vec_a = vdupq_n_u64(*reinterpret_cast<const uint64_t*>(a));
+                }
+
+                uint64x2_t vec_b;
+                if constexpr (!right_const) {
+                    vec_b = vld1q_u64(reinterpret_cast<const uint64_t*>(b) + i * 2);
+                } else {
+                    vec_b = vdupq_n_u64(*reinterpret_cast<const uint64_t*>(b));
+                }
+
+                uint8x16_t index = {0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1};
+                uint8x16_t mask = vqtbl1q_u8(loaded_mask, index);
```

Comment:
this piece of code is almost same exception the data_size parameter, it's better to extract the common part